data/service/api/v1/datasets_data_create.go

```diff
@@ -136,28 +137,26 @@ func DataSetsDataCreate(dataServiceContext dataService.Context) {
 // CollectProvenanceInfo from a request and its auth details.
 //
 // All work is done as a best effort right now.
-func CollectProvenanceInfo(ctx context.Context, req *rest.Request, details request.AuthDetails) *data.Provenance {
+func CollectProvenanceInfo(ctx context.Context, req *rest.Request, authDetails request.AuthDetails) *data.Provenance {
 	lgr := log.LoggerFromContext(ctx)
```

Comment:
From above, you'd add:
```suggestion
    if !authDetails.HasToken() {
        return nil
    }

	lgr := log.LoggerFromContext(ctx)
```